pkgs/development/compilers/llvm/README.md

```diff
@@ -77,3 +77,22 @@ Instead of applying the patches to the worktree per the above instructions, one
 For newer LLVM versions, enough has has been upstreamed,
 (see https://reviews.llvm.org/differential/query/5UAfpj_9zHwY/ for my progress upstreaming),
 that I have just assembled new gnu-install-dirs patches from the remaining unmerged patches instead of rebasing from the prior LLVM's gnu install dirs patch.
+
+## Adding a patch
+
+To add an LLVM patch in the Nixpkgs tree,
+
+1. Put the patch in the corresponding directory (`<VERSION>/<PACKAGE>`).
+
+   _Example_: If you want your patch to apply to clang version 12 (and, optionally, later versions), put it in `./12/clang`.
+2. Add the patch to the `patches` argument of the corresponding package in `./common`, guarded by a `lib.optional` or `lib.optionals` with the desired version constraints, passed through the `getVersionFile` function.
```

Comment:
```suggestion
2. Add the patch to the `patches` argument of the corresponding package in `./common`, guarded by a `lib.optionals` with the desired version constraints, passed through the `getVersionFile` function.
```

Some people get unhappy about `lib.optional`, so probably best to point people in the direction that's more likely to get through review easily.